### PR TITLE
[openthread] update OpenThread to fix ot-ctl reset

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

**Comment by jwhui** [comment]
@bukepo , a couple checks are currently failing. Thoughts?

**Comment by bukepo** [comment]
@jwhui I'm still working on it.

File: etc/docker/Dockerfile [hunk 1]
```diff
@@ -57,7 +57,7 @@ ENV OTBR_NORELEASE_DEPS \
 
 RUN apt-get update \
   && apt-get install -y $OTBR_DOCKER_REQS $OTBR_DOCKER_DEPS \
-  && git reset --hard && git clean -xffd \
+  && git clean -xffd \
```
Comment (jwhui) [inline etc/docker/Dockerfile:60]:
```suggestion
  && git clean -xfd \
```
Why `ff`?

Comment (bukepo) [reply etc/docker/Dockerfile]: it will also clean untracked files in sub modules.

**Pull request merged**